core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -146,10 +145,7 @@ public String getName() {
     @Override
     public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
         // make upstream queries ignore this query by returning `null`
-        // if either inner query builder is null or returns null-Query
-        if (positiveQuery == null || negativeQuery == null) {
-            return null;
-        }
+        // if either of inner query builders returns null-Query
```

Comment:
I wonder if these two lines of comment still make sense, maybe we could drop them?